File: src/main/java/io/github/tanguygab/tabadditions/bungee/TABAdditionsBungeeCord.java
package io.github.tanguygab.tabadditions.bungee;

import io.github.tanguygab.tabadditions.shared.TABAdditions;
import net.md_5.bungee.api.plugin.Plugin;

import java.util.HashMap;
import java.util.Map;

public final class TABAdditionsBungeeCord extends Plugin {

    private final Map<String,Boolean> servers = new HashMap<>();

    public TABAdditionsBungeeCord() {
        TABAdditions.addProperties();
    }

    public String getServerStatus(String server) {
        getProxy().getServers().get(server).ping((result, error) -> {
            if (error == null)
                servers.put(server,true);
            else servers.put(server,false);
        });
        return servers.containsKey(server) && servers.get(server) ? "Online" : "Offline";
    }

    @Override
    public void onEnable() {
        TABAdditions.setInstance(new TABAdditions(new BungeePlatform(this), this,getDataFolder()));
        TABAdditions.getInstance().load();
    }

    @Override
    public void onDisable() {
        TABAdditions.getInstance().disable();
    }

}

File: src/main/kotlin/io/github/tanguygab/tabadditions/bungee/TABAdditionsBungeeCord.kt
package io.github.tanguygab.tabadditions.bungee

import io.github.tanguygab.tabadditions.shared.TABAdditions
import net.md_5.bungee.api.ServerPing
import net.md_5.bungee.api.plugin.Plugin

class TABAdditionsBungeeCord : Plugin() {
    private val servers: MutableMap<String, Boolean> = HashMap()

    init {
        TABAdditions.addProperties()
    }

    fun getServerStatus(server: String): String {
        proxy.servers[server]!!.ping { _: ServerPing?, error: Throwable? ->
            servers[server] = error == null
        }
        return if (servers.getOrDefault(server, false)) "Online" else "Offline"
    }

    override fun onEnable() {
        TABAdditions.INSTANCE = TABAdditions(BungeePlatform(this), this, dataFolder)
        TABAdditions.INSTANCE.load()
    }

    override fun onDisable() {
        TABAdditions.INSTANCE.disable()
    }
}